### PR TITLE
Constants configuration

### PR DESCRIPTION
**Pull request by jduthon**

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -25,7 +26,7 @@ app.use((req, res, next) => {
   next()
 })
 
-app.set('port', process.env.PORT || 3000)
+app.set('port', process.env.PORT || config.port)
```
Comment (znarf) [inline lib/app.js]: What prevent us to skip process.env.PORT ?

File: lib/metrics.js [hunk 2]
```diff
@@ -12,6 +12,8 @@
 const { fromJS, Set, Map, isKeyed } = require('immutable')
 const { iso } = require('./util')
 const database = require('./database')
+const config = require('../config/variables')
+const { gc: gcConfig } = config.session
```
Comment (znarf) [inline lib/metrics.js]: To me, config.session.gc might be easier to read than gcConfig

**Comment by znarf** [comment]
There is something wrong about the naming, a variable can't be at the same time a constant.

File: lib/app.js (hunk 1 above)
Comment (jduthon) [reply lib/app.js]: retro-compatibility + the fact that `PORT` is a well known way to change port of a node app?
But I am not against removing it and only supporting the new way.

File: lib/metrics.js (hunk 2 above)
Comment (jduthon) [reply lib/metrics.js]: It's more typing but ultimately readability matters more so will fix

File: lib/app.js (hunk 1 above)
Comment (znarf) [reply lib/app.js]: So, with the rc configuration, the env variable would be `access_watch_port=80` not `PORT=80`.

Then, as discussed we keep `process.env.PORT`

**Comment by znarf** [comment]
@jduthon can you merge from master and make new constants from hub configurable?

File: dashboard/views/index.ejs [hunk 3]
```diff
@@ -20,7 +20,10 @@
 </div>
 
 <script data-api-base-url="<%= apiBaseUrl %>" data-websocket="<%= websocket %>" src="./main.js"></script>
+<script>
+  const config = JSON.parse('<%- uiConfig %>');
```
Comment (znarf) [inline dashboard/views/index.ejs]: Could that be written like that?

`const config = <%- uiConfig %>;`

Comment (jduthon) [reply dashboard/views/index.ejs]: No as uiConfig needs to be stringified before sent here as we need strings here, not JS.

**Pull request merged**